contracts/Vault.sol

```diff
@@ -0,0 +1,90 @@
+pragma solidity ^0.5.15;
+pragma experimental ABIEncoderV2;
+import { NameRegistry as Registry } from "./NameRegistry.sol";
+import { ParamManager } from "./libs/ParamManager.sol";
+import { IERC20 } from "./interfaces/IERC20.sol";
+import { Rollup } from "./Rollup.sol";
+import { RollupUtils } from "./libs/RollupUtils.sol";
+import { ITokenRegistry } from "./interfaces/ITokenRegistry.sol";
+import { Types } from "./libs/Types.sol";
+import { MerkleTreeUtils as MTUtils } from "./MerkleTreeUtils.sol";
+import { SpokeRegistry } from "./SpokeRegistry.sol";
+
+contract Vault {
+    Rollup public rollup;
+    Registry public nameRegistry;
+    MTUtils public merkleUtils;
+    SpokeRegistry public spokes;
+    ITokenRegistry public tokenRegistry;
+
+    constructor(address _registryAddr) public {
+        nameRegistry = Registry(_registryAddr);
+        rollup = Rollup(
+            nameRegistry.getContractDetails(ParamManager.ROLLUP_CORE())
+        );
+        tokenRegistry = ITokenRegistry(
+            nameRegistry.getContractDetails(ParamManager.TOKEN_REGISTRY())
+        );
+        merkleUtils = MTUtils(
```

Comment:
We don't have to use this on chain merkleUtils. We only use `verifyLeaf` here, so use MerkleTreeUtilsLib library is cheaper.
```
import { MerkleTreeUtilsLib } from "./MerkleTreeUtils.sol";
```

Reply:
Let me fix this in the next PR.